test/es-module/test-esm-loader-mock.mjs

```diff
@@ -0,0 +1,39 @@
+// Flags: --loader ./test/fixtures/es-module-loaders/mock-loader.mjs
+import '../common/index.mjs';
+import assert from 'assert/strict';
+import mock from 'node:mock';
+
+mock('node:events', {
+  EventEmitter: 'This is mocked!'
+});
+
+// This resolves to node:events
+assert.deepStrictEqual(await import('events'), Object.defineProperty({
```

Comment:
It `resolves to node:events`, you mean, it resolves to the mocked version?

Reply:
yes